api/src/main/java/org/pmiops/workbench/billing/BillingProjectBufferService.java

```diff
@@ -121,15 +121,22 @@ private DbBillingProjectBufferEntry makeCreatingBufferEntry() {
     final DbBillingProjectBufferEntry bufferEntry = new DbBillingProjectBufferEntry();
     bufferEntry.setFireCloudProjectName(createBillingProjectName());
     bufferEntry.setCreationTime(Timestamp.from(clock.instant()));
+    // Note: we set the lastSyncRequestTime column to the current timestamp as an optimization.
+    // If we leave this column as NULL, the sync process will prioritize this entry for immediate
+    // synchronization with Terra. Instead, we populate this column with the current timestamp
+    // since we know with high confidence that the initial status is CREATING. See RW-6192 for
+    // more context.
+    bufferEntry.setLastSyncRequestTime(Timestamp.from(clock.instant()));
```

Comment:
Nice find, this was a major bug. I was mistaken in my characterization of how this was supposed to work. There is no such ceiling on the number of creations in flight (possibly there should be..).